packages/block-editor/src/components/link-control/index.js

```diff
@@ -0,0 +1,249 @@
+	};
+
+	// Effects
+	const getSearchHandler = useCallback( ( value ) => {
+		const protocol = getProtocol( value ) || '';
+		const isMailto = protocol.includes( 'mailto' );
+		const isInternal = startsWith( value, '#' );
+		const isTel = protocol.includes( 'tel' );
+
+		const handleManualEntry = isInternal || isMailto || isTel || isURL( value ) || ( value && value.includes( 'www.' ) );
+
+		return ( handleManualEntry ) ? handleDirectEntry( value ) : handleEntitySearch( value );
+	}, [ handleDirectEntry, fetchSearchSuggestions ] );
+
+	// Render Components
+	const renderSearchResults = ( { suggestionsListProps, buildSuggestionItemProps, suggestions, selectedSuggestion, isLoading } ) => {
+		const resultsListClasses = classnames( 'block-editor-link-control__search-results', {
+			'is-loading': isLoading,
+		} );
+
+		const manualLinkEntryTypes = [ 'url', 'mailto', 'tel', 'internal' ];
+
+		return (
+			<div className="block-editor-link-control__search-results-wrapper">
+				<div { ...suggestionsListProps } className={ resultsListClasses }>
+					{ suggestions.map( ( suggestion, index ) => (
+						<LinkControlSearchItem
+							key={ `${ suggestion.id }-${ suggestion.type }` }
+							itemProps={ buildSuggestionItemProps( suggestion, index ) }
+							suggestion={ suggestion }
+							onClick={ () => onLinkChange( suggestion ) }
+							isSelected={ index === selectedSuggestion }
+							isURL={ manualLinkEntryTypes.includes( suggestion.type.toLowerCase() ) }
+							searchTerm={ inputValue }
+						/>
+					) ) }
+				</div>
+			</div>
+		);
+	};
+
+	return (
+		<Popover
+			className={ classnames( 'block-editor-link-control', className ) }
+			onClose={ closeLinkUI }
+			position="bottom center"
+			focusOnMount="firstElement"
+		>
+			<div className="block-editor-link-control__popover-inner">
+				<div className="block-editor-link-control__search">
+
+					{ ( ! isEditingLink && currentLink ) && (
+						<Fragment>
+							<p className="screen-reader-text" id={ `current-link-label-${ instanceId }` }>
+								{ __( 'Currently selected' ) }:
+							</p>
+							<div
+								aria-labelledby={ `current-link-label-${ instanceId }` }
+								aria-selected="true"
+								className={ classnames( 'block-editor-link-control__search-item', {
+									'is-current': true,
+								} ) }
+							>
+								<span className="block-editor-link-control__search-item-header">
+
+									<ExternalLink
+										className="block-editor-link-control__search-item-title"
+										href={ currentLink.url }
+									>
+										{ currentLink.title }
+									</ExternalLink>
+									<span className="block-editor-link-control__search-item-info">{ filterURLForDisplay( safeDecodeURI( currentLink.url ) ) || '' }</span>
+								</span>
+
+								<Button isDefault onClick={ startEditMode } className="block-editor-link-control__search-item-action block-editor-link-control__search-item-action--edit">
```

Comment:
Gutenberg uses a variation of [BEM](http://getbem.com/) where the one key difference is in how modifiers are applied, where Gutenberg expects that modifiers be applied with an additional class prefixed in some boolean sense (`is-foo`).

```suggestion
								<Button isDefault onClick={ startEditMode } className="block-editor-link-control__search-item-action is-edit">
```

See also, the CSS Naming guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/coding-guidelines.md#naming

Reply:
> Gutenberg uses a variation of [BEM](http://getbem.com/) where the one key difference is in how modifiers are applied, where Gutenberg expects that modifiers be applied with an additional class prefixed in some boolean sense (`is-foo`). The double-dash syntax from BEM is not standard.

See: #19738

Reply:
Hmm, I was sure I'd raised this during review, it looks this was marked as outdated and so not resolved - https://github.com/WordPress/gutenberg/pull/17846#discussion_r337853126

Should've really caught it in the final review. Cheers for spotting it.